xarray/core/groupby.py

```diff
@@ -66,6 +67,52 @@ def _dummy_copy(xarray_obj):
         raise AssertionError
     return res
 
+def _is_one_or_none(obj):
+    return obj == 1 or obj is None
```

Comment:
These slice objects are generated internally by groupby, so I think it's pretty far fetched to get `step is True`. Using `==` feels more idiomatic, anyways.